compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/MoveCompanionObjectFieldsLowering.kt

```diff
@@ -83,9 +84,15 @@ private class MoveCompanionObjectFieldsLowering(val context: CommonBackendContex
         companion.declarations.removeAll { it is IrAnonymousInitializer }
     }
 
-    private fun IrClass.allFieldsAreJvmField() =
-        declarations.filterIsInstance<IrProperty>()
-            .mapNotNull { it.backingField }.all { it.hasAnnotation(JVM_FIELD_ANNOTATION_FQ_NAME) }
+    private fun shouldExposeProperty(irClass: IrClass, irProperty: IrProperty): Boolean {
+        if (!irClass.isInterface && !irClass.isAnnotationClass)
```

Comment:
A little bit confused on this... I thought the property _should_ be moved from the companion object of the current class _is_ an interface or annotation class?

Reply:
```kotlin
if (!irClass.isJvmInterface)
```

Reply:
@punzki My understanding is that they should be moved into the parent class/interface whenever possible. A JVM specific limitation is that interfaces can only contain `public static final` fields so the checks have to be there. Does that make sense?

@sfs Thanks, that is neat.

Reply:
@ting-yuan That makes sense now, thanks for clarifying. I think a code comment could add clarity, since it's confusing why these particular conditions are used (especially the `!isJvmInterface`).

Reply:
Please take a look at the updated comments and let me know if it should be improved.